File: Bachelor/Semester4/Systems_for_Design_and_Implementation/Java/src/test/java/domain/validator/ReservationValidatorTest.java
package domain.validator;

import domain.Reservation;
import domain.Trip;
import domain.User;
import junit.framework.TestCase;

import java.time.LocalDateTime;

public class ReservationValidatorTest extends TestCase {

    private final ReservationValidator validator = new ReservationValidator();

    public void testValidate() {
        Trip trip = new Trip(1L, "Paris","Auto", LocalDateTime.MIN, 129, 20);
        User user = new User(1L,"Anna","Apples");
        Reservation reservation;
        reservation = new Reservation(1L, "Anne","072",20, trip, user);
        validator.validate(reservation);
        reservation = new Reservation(2L,"","",-4, null, user);
        try{
            validator.validate(reservation);
            fail();
        }catch (ValidationException e){
            String fullMessage = "Invalid client name\nInvalid phone number\nInvalid number of tickets\nInvalid trip\n";
            assertEquals(fullMessage, e.getMessage());
        }
        reservation = new Reservation(3L, "n", "076", 90, trip, user);
        try{
            validator.validate(reservation);
            fail();
        }catch (ValidationException e){
            assertEquals("Invalid number of tickets\n", e.getMessage());
        }
    }
}

File: Bachelor/Semester4/Systems_for_Design_and_Implementation/C#/ProiectCShMPP/Tests/domain/validator/ReservationValidatorTest.cs
using System;
using NUnit.Framework;
using ProiectCShMPP.domain;
using ProiectCShMPP.domain.validator;

namespace Tests.domain.validator
{
    [TestFixture]
    public class ReservationValidatorTest
    {
        private readonly IValidator<long, Reservation> _validator = new ReservationValidator();
        
        [Test]
        public void TestValidate()
        {
            User user = new User(1L)
            {
                Username = "Anne",
                Password = "anneMe"
            };
            Trip trip = new Trip(1L)
            {
                Destination = "Cairo",
                DepartureTime = new DateTime(2021, 1, 1, 1, 1, 1),
                Price = 120,
                Seats = 10,
                TransportFirm = "A .SRL"
            };
            Reservation reservation = new Reservation(1L)
            {
                Trip = trip,
                User = user,
                Client = "Anne",
                PhoneNumber = "082",
                Tickets = 1
            };
            _validator.Validate(reservation);
            try
            {
                _validator.Validate(new Reservation(2L)
                {
                    Client = "",
                    PhoneNumber = "",
                    Tickets = 19000,
                    Trip = trip,
                    User = user
                });
                Assert.Fail();
            }
            catch (ValidationException e)
            {
                Assert.AreEqual("Invalid client name\nInvalid phone number\nInvalid number of tickets\n", e.Message);
            }
        }
    }
}